README.md

```diff
@@ -72,3 +72,101 @@ The following links describe the structure of the system and detail some of the
+A BREAKING CHANGE can be part of commits of any _type_.
+1. Others: commit _types_ other than `fix:` and `feat:` are allowed, for example [commitlint-config-conventional](https://github.com/marionebl/commitlint/tree/master/%40commitlint/config-conventional) (based on the [the Angular convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines)) recommends `chore:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others.
+We also recommend `improvement` for commits that improve a current implementation without adding a new feature or fixing a bug.
+Notice these types are not mandated by the conventional commits specification, and have no implicit effect in semantic versioning (unless they include a BREAKING CHANGE).
+<br />
+A scope may be provided to a commit's type, to provide additional contextual information and is contained within parenthesis, e.g., `feat(parser): add ability to parse arrays`.
+
+## Examples
+
+### Commit message with description and breaking change in body
+```
+feat: allow provided config object to extend other configs
+
+BREAKING CHANGE: `extends` key in config file is now used for extending other config files
+```
+
+### Commit message with no body
+```
+docs: correct spelling of CHANGELOG
+```
+
+### Commit message with scope
+```
+feat(lang): added polish language
+```
+
+### Commit message for a fix using an (optional) issue number.
+```
+fix: minor typos in code
+
+see the issue for details on the typos fixed
+
+fixes issue #12
+```
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+1. Commits MUST be prefixed with a type, which consists of a noun, `feat`, `fix`, etc., followed by a colon and a space.
+1. The type `feat` MUST be used when a commit adds a new feature to your application or library.
+1. The type `fix` MUST be used when a commit represents a bug fix for your application.
+1. An optional scope MAY be provided after a type. A scope is a phrase describing a section of the codebase enclosed in parenthesis, e.g., `fix(parser):`
+1. A description MUST immediately follow the type/scope prefix.
+The description is a short description of the code changes, e.g., _fix: array parsing issue when multiple spaces were contained in string._
+1. A longer commit body MAY be provided after the short description, providing additional contextual information about the code changes. The body MUST begin one blank line after the description.
+1. A footer MAY be provided one blank line after the body.
+  The footer SHOULD contain additional issue references about the code changes (such as the issues it fixes, e.g.,`Fixes #13`).
```

Comment:
Currently we mandate the JIRA issue number must be within the 1st line of the commit message, so that gitbot can automatically associate it with the issue.

According to these guidelines, we should reference the issue in the footer.
@GordonSmith - do these footer issue references have a format specification so that the automatic generation of changelogs etc. see them?
@richardkchapman - can our gitbot be changed to look for the JIRA in the footer?

Reply:
@GordonSmith - in fact, which JIRA is this PR associated with?

Reply:
See my first comment fort the commit message - it included:
```
Fixes #20977
```

FYI - here is what the generated changelog looks like:  https://github.com/GordonSmith/HPCC-Platform/blob/STANDARD_VERSION_TEST/CHANGELOG.md

You can see the hyperlink back to issue is correct...

Reply:
Ah I see it now, I missed it because normally the body of our commit messages are together with the 1st comment and above the generated checklist.

Reply:
That part is only different as I needed to rebase after I issued the orig PR...